awq/quantize/quantizer.py

```diff
@@ -228,6 +238,32 @@ def _apply_quant(self, module, named_linears: Dict[str, nn.Linear]):
             set_op_by_name(module, name, q_linear)
             clear_memory()
 
+    @torch.no_grad()
+    def _module_forward(
+        self, x: torch.Tensor, module: torch.nn.Module, module_kwargs: Dict
+    ) -> torch.Tensor:
+        if self.n_parallel_calib_samples is None:
+            # runs through all samples at once
+            module_output = module(x, **module_kwargs)
+            if isinstance(module_output, tuple):
+                module_output = module_output[0]
+        else:
+            # memory efficiently runs through all calibration samples
+            # but only n_parallel_calib_samples at a time
+            module_output = []
+            for i in range(0, x.shape[0], self.n_parallel_calib_samples):
+                x_partial = x[i : i + self.n_parallel_calib_samples]
+                partial_output = module(x_partial, **module_kwargs)
+
+                if isinstance(partial_output, tuple):
+                    partial_output = partial_output[0]
+
+                module_output.append(partial_output)
+
+            module_output = torch.cat(module_output, dim=0)
```

Comment:
OOM often happens here. Maybe you need move partial_output to cpu by module_output.append(partial_output.detach.cpu())

Reply:
yep, when I quantize qwen1.5-32B with seq_len=32k、max_calib_samples=128 and n_parallel_calib_samples=1，GPU OOM happens. I move inp_flat and partial_output to cpu, then everything seems Ok

Reply:
I added .cpu() for now.